File: software_challenge_sdk/src/server_api/sc/protocol/responses/GameRoomMessage.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

/**
 * Send to all administrative clients after a playerClient joined a game via a JoinRoomRequest
 */

@XStreamAlias(value="joinedGameRoom")
public class GameRoomMessage extends ProtocolErrorMessage {

  @XStreamAsAttribute
  private String roomId;

  @XStreamAsAttribute
  private boolean existing;

  public GameRoomMessage(String roomId, boolean existing) {
    this.roomId = roomId;
    this.existing = existing;
  }

}

File: socha-sdk/src/server-api/sc/protocol/responses/GameRoomMessage.kt
package sc.protocol.responses

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Sent to all administrative clients after a playerClient joined a game via a JoinRoomRequest. */
@XStreamAlias(value = "joinedGameRoom")
data class GameRoomMessage(
        @XStreamAsAttribute
        private val roomId: String,
        @XStreamAsAttribute
        private val existing: Boolean
): ProtocolMessage
